content/blog/intro-to-gamedev/index.md

```diff
@@ -0,0 +1,560 @@
+
+At this point, however, we actually have 2 types of objects that interact with the world: `Player` and `ScrollingObject`. But it makes sense for us to coalesce all instances in one list of objects to loop over. If we were to go in an [object-oriented](https://en.wikipedia.org/wiki/Object-oriented_programming) direction, we could create a class hierarchy that perhaps could make both our classes inherit from a common one that had all the shared functionality and data. Another good approach would be to create an interface that enforced the methods/fields that they both share. However, since we are using TypeScript and it has a [very powerful type system](https://www.typescriptlang.org/docs/handbook/advanced-types.html#union-types), we can try to implement one of the lesser pragmatic options by creating a Union type and an alias for it. Remember, we are trying to go for low friction solutions for our little project.
+
+So let's go ahead and implement all this on our `GameApp` class.:
+
+```ts
+// creating an alias for our Union type that we can use
+type WorldObjects = Player | ScrollingObject
+
+export class GameApp {
+  public app: PIXI.Application
+
+  static PressedSpace: boolean = false
+  static Stage: PIXI.Container
+  static ActiveEntities: Array<WorldObjects> = new Array<WorldObjects>()
+  static GameOver: boolean = false
+  static ScrollSpeed: number = 3
+
+  static GroundPosition: number = 0 // ground position, given by screen height
+  static Width: number = 0 // width of game screen, given by screen width
+
+  // score of current run
+  static Score: number = 0
+
+  // max score achieved in session
+  static MaxScore: number = 0
+
+  // next score in which we should place an obstacle
+  static ScoreNextObstacle: number = 0
+}
+```
+
+Objects whose types are of this Union type (`Player | ScrollingObject`) have accessible members that are defined both on our Player and on ScrollingObject types. As seen above, I created a new static member called `ActiveEntities` which is an array of `WorldObject`s. We now have to change `GameApp.Update()` to loop over this array.
+
+But before that, you might have noticed as well that I created a couple more variables for tracking score stuff. This will allow us to do a couple of things:
+
+- Keep score for the current run
+- Save the maximum score achieved in a session
+- Know when the next obstacle should spawn (each time we create one, we set the score at which the next one should appear)
+
+To keep the code simple for now, everytime we spawn an obstacle we'll spawn a cloud for aesthetics. There's no real reason why we couldn't add a more random element to these decorations later, but for now this will keep the code easier to study/describe.
+
+Keeping that in mind, let's see what our current game update looks like:
+
+```ts
+    static Update(delta: number) {
+
+        // if we haven't lost yet let's update everything, otherwise wait for spacebar press to restart game
+        if (!this.GameOver)
```

Comment:
This makes it even more confusing, why sometimes are you using `this` to refer to the class `GameApp` and sometimes you are using `GameApp` to refer to it